common/src/main/java/com/pg85/otg/generator/resource/LiquidGen.java

```diff
@@ -85,6 +86,11 @@ public void spawn(LocalWorld world, Random rand, boolean villageInChunk, int x,
 
         if (!world.isNullOrAir(x, y, z, false) && (!sourceBlocks.contains(world.getMaterial(x, y, z, false))))
             return;
+        
+        if (!bLoaded ) {
```

Comment:
Could put this in a method in Resource instead, no repeating code for each Gen class?